utils/task_utils.go

```diff
@@ -23,86 +23,101 @@ func checkTaskFinished(task *models.Task) int8 {
 	}
 }
 
-func WaitTask(client *apiclient.Cloudtower, id *string) error {
+func WaitTask(ctx context.Context, client *apiclient.Cloudtower, id *string, interval time.Duration) error {
+	if ctx == nil {
```

Comment:
not sure if this should be implemented, use nil as a context input param will show a warning in IDE

Reply:
I think most of the go API designs use a **required** context params, user can pass todo or background context if they do not need the timeout feature

Reply:
removed nil context handling